jimp.d.ts

```diff
@@ -30,285 +166,413 @@ declare namespace Jimp {
             cb?: Jimp.ImageCallback
         );
 
-        bitmap: Bitmap;
-
-        clone(cb?: Jimp.ImageCallback): Jimp;
-        quality(n: number, cb?: Jimp.ImageCallback): this;
-        deflateLevel(l: number, cb?: Jimp.ImageCallback): this;
-        deflateStrategy(s: number, cb?: Jimp.ImageCallback): this;
-        filterType(f: number, cb?: Jimp.ImageCallback): this;
-
-        rgba(bool: boolean, cb?: Jimp.ImageCallback): this;
-        background(hex: number, cb?: Jimp.ImageCallback): this;
-        scan(
-            x: number,
-            y: number,
-            w: number,
-            h: number,
-            f: (x: number, y: number, idx: number) => any,
-            cb?: Jimp.ImageCallback
-        ): this;
-        getWidth(): number;
+        // Methods
         getHeight(): number;
+        getWidth(): number;
+        inspect(): string;
+        toString(): string;
         getMIME(): string;
         getExtension(): string;
-        getPixelIndex(
+        write(path: string, cb?: Jimp.ImageCallback): this;
+        writeAsync(path: string): Promise<Jimp>;
+        deflateLevel(l: number): this;
+        deflateLevel<T>(l: number, cb: Jimp.ImageCallback<T>): T;

```

Comment:
Why does some of the have `<T>`? What is the purpose

Reply:
If you look at the code, at the end of those function there is an if statement to check whether you passed a callback. If not, the function returns the Jimp instance for chaining, otherwise it returns the result of the callback call. The generic argument is inferred from the return type of the callback